### PR TITLE
Gui bug

### PR DESCRIPTION
**Pull request by bart-de-jonge**
Fixed bug where dummypane didn't dissapear
Fixed bug with feedbackpane
Refactored resizing code

### CODE REVIEW

File: src/main/java/control/TimelineController.java [hunk 1]
```diff
@@ -53,10 +53,9 @@ public void addCameraShot(int cameraIndex, String name, String description,
         CameraShot newShot = new CameraShot(name,description, startCount, endCount);
         this.scriptingProject.getCameraTimelines().get(cameraIndex).addShot(newShot);
         CameraShotBlock shotBlock = new CameraShotBlock(newShot.getInstance(), cameraIndex,
-                                                        rootPane.getRootCenterArea(),
-                                                        startCount, endCount);
+                rootPane.getRootCenterArea(), startCount, endCount,
+                this.rootPane.getRootCenterArea().getGrid());
```
Comment (geenen124) [inline src/main/java/control/TimelineController.java]: Why is the timeline grid being passed to the constructor if the RootCenterArea is already being passed?

File: src/main/java/gui/CameraShotBlock.java [hunk 2]
```diff
@@ -24,13 +26,15 @@
...
      * @param endCount - the end count of this shot
+     * @param grid - the grid this camerashotblock should belong to
      */
     public CameraShotBlock(int shotId, int timetableNumber, RootCenterArea rootCenterArea,
-                           double beginCount, double endCount) {
+                           double beginCount, double endCount, TimelinesGridPane grid) {
         super(rootCenterArea, beginCount, endCount);
         this.shotId = shotId;
         this.timetableNumber = timetableNumber;
         thisBlock = this;
```
Comment (Markvanderuit) [inline src/main/java/gui/CameraShotBlock.java:35]: Couldn't we just use 'this' wherever the variable thisBlock is being used? Probably has a good reason, but I'm curious.

Comment (bart-de-jonge) [inline src/main/java/gui/CameraShotBlock.java:35]: This is to be able to access the block from within event handlers

File: src/main/java/gui/TimetableBlock.java [hunk 3]
```diff
@@ -210,14 +220,14 @@ private boolean snapPane(Region targetRegion, Pane mappingPane,
...
             xCoordinate = mappingPane.getLayoutX() + mappingPane.getWidth() / 2;
             yCoordinate++;
-            System.out.println(yCoordinate);
         }
 
         SnappingPane myPane = pane.getGrid().getMyPane(xCoordinate, yCoordinate);
         if (myPane != null) {
-            int numCounts = (int) Math.round(dummyPane.getHeight() / pane.getCountHeight());
+            int numCounts = (int) Math.round(mappingPane.getHeight() / pane.getCountHeight());
+            System.out.println(numCounts);
```
Comment (geenen124) [inline src/main/java/gui/TimetableBlock.java]: Debug statement in the gui method?

File: src/main/java/gui/TimetableBlock.java [hunk 4]
```diff
@@ -195,8 +204,9 @@ private void onMouseDraggedHelper(MouseEvent event) {
      * @param dragType - The type of drag used while snapping (move, resize)
      * @return - boolean that indicates if the snap was possible and completed
      */
-    private boolean snapPane(Region targetRegion, Pane mappingPane,
+    private boolean snapPane(Region targetRegion, Region mappingPane,
                              double x, double y, DraggingTypes dragType) {
+        System.out.println(mappingPane.toString());
```
Comment (geenen124) [inline src/main/java/gui/TimetableBlock.java]: Another debug print statement

**Comment by Markvanderuit** [comment]
Looks good, and tested for running. Might want to make blocks visible again, and fix some of the feedback. Afterwards, I think we can merge.

**Comment by geenen124** [comment]
Bug squashing ftw!

**Pull request merged**